internal/controller/datadogagent/component/agent/default.go

```diff
@@ -353,11 +354,15 @@ func initContainers(dda metav1.Object, requiredContainers []apicommon.AgentConta
 }
 
 func agentSingleContainer(dda metav1.Object) []corev1.Container {
+	autopilot := false
+	if ddaTyped, ok := dda.(*v2alpha1.DatadogAgent); ok {
```

Comment:
What do you think about moving the autopilot logic to `internal/controller/datadogagent/global/global.go`? Many functions there already have access to the `dda.Spec.Global` config object so you can avoid performing type assertions

https://github.com/DataDog/datadog-operator/blob/main/internal/controller/datadogagent/global/global.go#L259-L276